workshops/views.py

```diff
@@ -952,3 +1008,24 @@ def _get_pagination_items(request, all_objects):
             result = paginator.page(paginator.num_pages)
 
     return result
+
+
+def _time_series(request, data, title):
+    '''Prepare time-series data for display and render it.'''
+
+    # Make sure addition will work.
+    for key in data:
+        if data[key] is None:
+            data[key] = 0
+
+    # Create running total.
+    data = list(data.items())
+    data.sort()
```

Comment:
Alternatively to this sorting we could use `.order_by` in the queries above, and use [OrderedDict](https://docs.python.org/3/library/collections.html#collections.OrderedDict) instead of `dict`.